src/Engine/Utils.java

```diff
@@ -527,8 +520,8 @@ public static void sortLocationsByTravelTime(Unit unit, ArrayList<XYCoord> mapLo
     if( map.isLocationValid(x, y) )
     {
       int range = (piercing)? viewer.model.visionRangePiercing : viewer.model.visionRange;
-      // if it's a surface unit, give it the boost the terrain would provide
-      if( viewer.model.isSurfaceUnit() )
+      // if it's a surface unit, give it the boost the terrain would provide, so long as it's not adjacent-only vision
+      if( (!piercing || range > 1) && viewer.model.isSurfaceUnit() )
```

Comment:
Why `!piercing || range > 1`? Are there any units with piercing vision range > 1? Why should they get the piercing vision boost?

Reply:
`!piercing` because that's normal vision. If I got a rocket (vision 1) onto a mountain I'd expect to see 3 spaces.
`range > 1` because everybody gets range 1 piercing vision and they don't get a piercing boost from mountains; by contrast, when your whole vision is piercing then you *do* get a piercing vision boost.

Reply:
Are there units with only piercing vision? Also, doesn't this exclude the possibility of units with only piercing vision that can see one space?

Reply:
Not at the moment, no. I guess that's a possibility.
Sonja/Lin are implemented to make piercing == normal after the buff.

It does exclude the possibility of units that *should* be buffed with complete piercing vision that only have 1 vision, but in practice, piercing vision only comes with a general vision buff.

Reply:
All right, sustained. Mostly because I know it would probably be a silly amount of work to force it to behave consistently, and as you say this should be fine in practice.